express/src/controllers/manage-account.ts

```diff
@@ -84,24 +196,22 @@ export const processUpdateServiceForm = async function (req: Request, res: Respo
         return;
     }
 
-    const userId = AuthenticationResultParser.getCognitoId(req.session.authenticationResult as AuthenticationResultType);
-    if (userId === undefined) {
-        console.log("Can't get CognitoId from authenticationResult in session");
-        res.render("there-is-a-problem.njk");
-        return;
-    }
-
+    const selfServiceClientId = req.params.selfServiceClientId;
+    const clientId = req.params.clientId;
     const s4: SelfServiceServicesService = req.app.get("backing-service");
+
+    // TODO service_name is the db layer leaking into the domain
+    // TODO is this correct? Don't we also need to update the pk:service sk:service entry? We need an updateService call
```

Comment:
The lambda takes care of all of this via a step-function.  The frontend wants to update the client, the backend does that and hides away the complexity that using DynamoDB introduces.  This is probably arse about face because the client name and service name were once the same.  Which probably makes your change that hardcodes 'integration' interesting!

Reply:
When updating the service name we need to update the service name field on the client (pk:service sk:client) and on the corresponding pk:service sk:service entry. I don't believe the current handler do that. Changing the service name is out of scope for the MVP so that's fine for now but we'll need to revisit the lambdas and the APIs.

The client name is just for tidyiness as at the moment we don't let the users set or change the client name. They're allowed to have an 'integration' client and in the future a 'production'. We don't have to worry about this until we get to the multi-client functionality.